pkg/kubelet/nodestatus/setters.go

```diff
@@ -90,17 +90,16 @@ func NodeAddress(nodeIP net.IP, // typically Kubelet.nodeIP
 			if nodeIP != nil {
 				enforcedNodeAddresses := []v1.NodeAddress{}
 
-				var nodeIPType v1.NodeAddressType
+				nodeIPTypes := make(map[v1.NodeAddressType]bool)
 				for _, nodeAddress := range nodeAddresses {
 					if nodeAddress.Address == nodeIP.String() {
 						enforcedNodeAddresses = append(enforcedNodeAddresses, v1.NodeAddress{Type: nodeAddress.Type, Address: nodeAddress.Address})
-						nodeIPType = nodeAddress.Type
-						break
+						nodeIPTypes[nodeAddress.Type] = true
```

Comment:
Right. Before the `break` was introduced this loop appended both external and internal ip address with `ip == nodeIP.String()`. After the `break`, the second loop in `if len(enforcedNodeAddresses) > 0` branch actually appended all internal ip addresses (not just `nodeIP`) which comes in random order so instead of returning `(external nodeIP, internal nodeIP, hostname)`, the `enforcedNodeAddresses` now returns `(external nodeIP, internal nodeIP, internal ip1, ...)` or `(external nodeIP, internal ip1, internal nodeIP, ...)`. This is quite a regression.